File: test_samples/valid/no_single_use_functions_same_file.js
function reusedAbstraction(foo) {
  return bar(foo);
}

function firstConsumer(input) {
  return reusedAbstraction(input);
}

function secondConsumer(input) {
  return reusedAbstraction(input);
}

function recursiveFunction(value) {
  if (value <= 0) return 0;
  return recursiveFunction(value - 1);
}

function sameBlockAbstraction(foo) {
  return bar(foo);
}

function sameBlockConsumer(input) {
  const first = sameBlockAbstraction(input);
  return sameBlockAbstraction(first);
}

function nestedAbstraction(foo) {
  return bar(foo);
}

function nestedConsumer(input) {
  return nestedAbstraction(nestedAbstraction(input));
}

File: test_samples/valid/no_single_use_functions_same_file.py
def reused_abstraction(foo: str) -> int:
    return bar(foo)


def first_consumer(input_value: str) -> int:
    return reused_abstraction(input_value)


def second_consumer(input_value: str) -> int:
    return reused_abstraction(input_value)


def recursive_function(value: int) -> int:
    if value <= 0:
        return 0
    return recursive_function(value - 1)


def same_block_abstraction(foo: str) -> int:
    return bar(foo)


def same_block_consumer(input_value: str) -> int:
    first = same_block_abstraction(input_value)
    return same_block_abstraction(first)


def nested_abstraction(foo: str) -> int:
    return bar(foo)


def nested_consumer(input_value: str) -> int:
    return nested_abstraction(nested_abstraction(input_value))
